### PR TITLE
Feature/sign in

### PR DESCRIPTION
**Pull request by Izat150487**

### CODE REVIEW

File: src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java [hunk 1]
```diff
@@ -16,8 +17,12 @@ public class AuthController {
...
 
-    @PostMapping("register")
-    public AuthenticationResponse register(@RequestBody RegisterRequest request) {
-        return authService.register(request);
-    }
+  @PostMapping("authenticate")
+  public AuthenticationResponse authenticate(@RequestBody AuthenticationRequest request){
+    return authService.authenticate(request);
+  }
+  @PostMapping("register")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java:24]: Next line

File: src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationRequest.java [hunk 2]
```diff
@@ -1,12 +1,15 @@
 package us.peaksoft.gadgetarium.dto;
 
-import lombok.*;
-
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+@Getter
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationRequest.java:8]: Next line

File: src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java [hunk 3]
```diff
@@ -1,35 +1,26 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 import us.peaksoft.gadgetarium.enums.Role;
-
 import java.time.LocalDate;
-
+@Getter
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java:10]: Next line

File: src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java [hunk 4]
```diff
@@ -1,35 +1,26 @@
...
-    private String password;
-
-    private String phoneNumber;
-
-    private Role role;
-
-    private LocalDate createdDate;
-
-}
+  private String token;
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java:16]: Next line

File: src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java [hunk 5]
```diff
@@ -32,12 +31,12 @@ public AuthenticationProvider authenticationProvider(){
         return authProvider;
     }
     @Bean
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/AuthConfiguration.java:33]: Next line

File: src/main/java/us/peaksoft/gadgetarium/security/JwtAuthFilter.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import java.io.IOException;
+@Component
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/JwtAuthFilter.java:18]: Next line

File: src/main/java/us/peaksoft/gadgetarium/security/JwtService.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+@Service
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/JwtService.java:16]: Next line

File: src/main/java/us/peaksoft/gadgetarium/security/JwtService.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        return claimsResolver.apply(claims);
+    }
+    public String generateToken(UserDetails userDetails){
+        return generateToken(new HashMap<>(), userDetails);
+    }
+    public String generateToken(Map<String,Object> extraClaims, UserDetails userDetails ){
+        return Jwts.builder().setClaims(extraClaims)
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis()+ 1000 * 60 * 24))
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/JwtService.java:35]: Change to one month

File: src/main/java/us/peaksoft/gadgetarium/security/JwtService.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                .signWith(getSignInKey(), SignatureAlgorithm.HS256)
+                .compact();
+    }
+    public boolean isTokenValid(String token, UserDetails userDetails){
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/security/JwtService.java:39]: Next line

File: src/main/java/us/peaksoft/gadgetarium/service/DiscountService.java [hunk 10]
```diff
@@ -22,3 +22,7 @@ public interface DiscountService {
     List<ProductResponse> getProductsByDiscountId(Long id, int page, int size);
 
 }
+
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/service/DiscountService.java:25]: Pls remove lines 25 to 28

**Review by i-Coderr — COMMENTED** [review]
Commented at 11:31 pm 07/04/23

File: src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationRequest.java (hunk 2 above)
Comment (Izat150487) [reply src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationRequest.java:8]: fixed all wrong lines..

File: src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java (hunk 1 above)
Comment (Izat150487) [reply src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java:24]: fixed all wrong lines

File: src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java (hunk 4 above)
Comment (Izat150487) [reply src/main/java/us/peaksoft/gadgetarium/dto/AuthenticationResponse.java:16]: fixed all wrong lines

File: src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java [hunk 11]
```diff
@@ -19,6 +20,11 @@ public class AuthController {
 
     private final AuthService authService;
 
+    @PostMapping("login")
+    public AuthenticationResponse authenticate(@RequestBody AuthenticationRequest request) {
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/AuthController.java:24]: Add swagger description

**Review by i-Coderr — APPROVED** [review]
Almost done, good job!

**Pull request merged**